src/town_gui.cpp

```diff
@@ -1293,6 +1287,23 @@ struct FoundTownWindow : Window {
 		}
 	}
 
+	void OnQueryTextFinished(std::optional<std::string> str) override
+	{
+		/* Was 'cancel' pressed? */
+		if (!str.has_value()) return;
+
+		if (str->empty()) return;
+		uint value = strtoul(str->c_str(), nullptr, 10);
```

Comment:
#14160 was merged. `strtoul` is no longer valid.
```suggestion
		auto value = ParseInteger(*str);
		if (!value.has_value()) return;
```